### PR TITLE
Handle errors when ignoring failing sources

### LINKED ISSUE #6387: Unhelpful exception with RestoreCommand
While using the API for restore, I'm trying to install a package that doesn't exist. Also one of the sources doesn't exist. Restore fails with an unhelpful exception. [Details](https://gist.github.com/lbargaoanu/17344fcecda10db2b4b9aff46c27185a).
Version 4.05.0.4685.

### LINKED ISSUE #7036: Package reference restore fails
If I add an invalid V3 source the restore fails with an unhandled exception when there's no exact match. IgnoreFailedSources is true. For example the invalid source can be "http://bad.json".
It reproduces both in Visual Studio 2017 15.7.3 and in the restore API 4.7.0.
[The exact error](https://gist.github.com/lbargaoanu/ce0ff639c7df740bea01e537def3ea13).

### PR DESCRIPTION
**Pull request by lbargaoanu**
Fixes NuGet/Home#7036, fixes NuGet/Home#6387. Some overlap with #2307, but I've only noticed after opening the PR :)

### CODE REVIEW

**Comment by rohit21agrawal** [comment]
@lbargaoanu can you rebase from dev so we can run all our tests against this change? we made some changes recently that are required in your branch to run certain tests.

**Comment by lbargaoanu** [comment]
Rebased.

**Comment by lbargaoanu** [comment]
Rebase with the fix from #2307. I'm not sure what exactly fails on the CI build, but locally, with CI true, the tests seem to pass.

**Comment by rohit21agrawal** [comment]
i'll paste the names of tests that fail on CI that might be related to your change. let me re-rnu again.

**Comment by rohit21agrawal** [comment]
the tests are fine, the ones that failed are known to be flaky.

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/SourceRepositoryDependencyProvider.cs [hunk 1]
```diff
@@ -546,5 +539,12 @@ private void AddNuspecCache(FindPackageByIdResource resource)
             return packageVersions;
         }
 
+        private async Task LogWarning(string id, FatalProtocolException e)
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/SourceRepositoryDependencyProvider.cs]: LogWarningAsync

**Review by nkolev92 — APPROVED** [review]
LGTM  👍 

just a minor naming comment, kudos on the tests. 

Thanks for the contribution @lbargaoanu

**Comment by lbargaoanu** [comment]
Renamed. Thanks for the feedback.

**Review by rohit21agrawal — APPROVED** [review]

**Pull request merged**